ab2d-aggregator/src/main/java/gov/cms/ab2d/aggregator/Aggregator.java

```diff
@@ -0,0 +1,283 @@
+        List<File> bestFiles = getBestFiles(fileType);
+        if (bestFiles.isEmpty()) {
+            return NOT_PERFORMED;
+        }
+        String fileName = getNextFileName(fileType);
+        if (fileName == null) {
+            return NOT_PERFORMED;
+        }
+        combineFiles(bestFiles, fileName);
+        cleanUpFiles(bestFiles);
+        return AggregatorResult.PERFORMED;
+    }
+
+    @SuppressWarnings("PMD.DataflowAnomalyAnalysis")
+    void removeEmptyFiles() {
+        String fDir = this.mainDirectory + File.separator + this.finishedDir;
+        List<File> availableFiles = listFiles(fDir, DATA);
+        availableFiles.addAll(listFiles(fDir, ERROR));
+        List<File> emptyFiles = availableFiles.stream()
+                .filter(f -> {
+                            long size;
+                            try {
+                                size = getSizeOfFileOrDirectory(f.getAbsolutePath());
+                            } catch (IOException e) {
+                                size = 0L;
+                            }
+                            return size == 0;
+                        }).collect(Collectors.toList());
+        emptyFiles.forEach(File::delete);
+    }
+
+    /**
+     * Return the name of the next file to create given it is or isn't an error file and the
+     * last file index
+     *
+     * @param type - type of output file
+     * @return the name of the next file to aggregate into
+     */
+    String getNextFileName(FileOutputType type) {
+        switch (type) {
+            case ERROR:
+                return Path.of(mainDirectory, getNextErrorFileName()).toFile().getAbsolutePath();
+            case DATA:
+                return Path.of(mainDirectory, getNextDataFileName()).toFile().getAbsolutePath();
+            default:
+                return null;
+        }
+    }
+
+    /**
+     * If we should aggregate files
+     *
+     * @param type - type of file
+     * @return - true if we have enough files or the worker is done writing out files
+     */
+    boolean okayToDoAggregation(FileOutputType type) {
+        long size = getSizeOfFiles(this.mainDirectory + File.separator + this.finishedDir, type);
+        return (size > ((long) this.multiplier * getMaxFileSize())) || isJobDoneStreamingData();
+    }
+
+    String getNextDataFileName() {
+        String partName = Integer.toString(currentFileIndex);
+        var paddedPartitionNo = StringUtils.leftPad(partName, 4, '0');
+        currentFileIndex++;
+        return contractNumber +
+                "_" +
+                paddedPartitionNo +
+                DATA.getSuffix();
+    }
+
+    String getNextErrorFileName() {
+        String partName = Integer.toString(currentErrorFileIndex);
+        var paddedPartitionNo = StringUtils.leftPad(partName, 4, '0');
+        currentErrorFileIndex++;
+        return contractNumber +
+                "_" +
+                paddedPartitionNo +
+                ERROR.getSuffix();
+    }
```

Comment:
In my add_aggregator_enum branch, I made this an instance method within the enum looking like: 
```
public String namePart(int[] fileIndexes, String contractNumber) {
        String partName = Integer.toString(fileIndexes[ordinal()]);
        var paddedPartitionNo = StringUtils.leftPad(partName, 4, '0');
        fileIndexes[ordinal()]++;
        return contractNumber +
                "_" +
                paddedPartitionNo +
                getSuffix();
    }
```
To initialize the array, I put:
`        Arrays.fill(fileIndexes, 1);
`
In the constructor.

Reply:
I did something similar.